hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -256,12 +256,9 @@ private ContainerWithPipeline getContainerWithPipelineCommon(
     }
 
     if (pipeline == null) {
```

Comment:
I think it's not safe design to pass the COntainerReplicas directly to pluggable pipeline creator classes.

Reply:
For what reason? Is it because the replicas are mutable right now? If so, there is already a PR to fix that on master #3071, as it is a wider problem.

Reply:
Yes. But I see your PR discussing the fundamental issue of that. Since we track down that in master, we can ahead with this change here. Post that PR, I assume that will be a separate set than original one.